src/DotVVM.Tools.StartupPerfTester/StartupPerformanceTest.cs

```diff
@@ -0,0 +1,168 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Threading;
+using Medallion.Shell;
+
+namespace DotVVM.Tools.StartupPerfTester
+{
+    public class StartupPerformanceTest
+    {
+        private readonly FileInfo project;
+        private readonly TestTarget type;
+        private readonly int repeat;
+        private readonly string url;
+        private readonly bool verbose;
+        private readonly int timeout;
+
+        public StartupPerformanceTest(FileInfo project, TestTarget type, int repeat, string url, bool verbose, int timeout)
+        {
+            this.project = project;
+            this.type = type;
+            this.repeat = repeat;
+            this.url = url;
+            this.verbose = verbose;
+            this.timeout = timeout;
+        }
+
+
+        public void HandleCommand()
+        {
+            // test project existence
+            var projectPath = project.FullName;
+            if (!project.Exists)
+            {
+                throw new Exception($"The project {projectPath} doesn't exist!");
+            }
+
+            // find a random port
+            var port = NetworkingHelpers.FindRandomPort();
+            var urlToTest = $"http://localhost:{port}/{url.TrimStart('/')}";
+
+            // prepare directories
+            var dir = Path.GetDirectoryName(projectPath);
+            TraceOutput($"Project dir: {dir}");
+            var tempDir = FileSystemHelper.CreateTempDir();
+            TraceOutput($"Temp dir: {tempDir}");
+
+            long measuredTime = 0;
+            if (type == TestTarget.Owin)
+            {
+                // OWIN
+                TraceOutput($"Publishing...");
+                RunProcessAndWait(@"c:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe",
```

Comment:
Can't we use [vswhere](https://www.nuget.org/packages/Microsoft.VisualStudio.Setup.Configuration.Interop) instead of hard-coding the path?